src/ray/rpc/gcs_server/gcs_rpc_client.h

```diff
@@ -225,14 +229,42 @@ class GcsRpcClient {
   }
 
   void Reconnect() {
+    absl::MutexLock lock(&mutex_);
     if (get_server_address_) {
-      auto address = get_server_address_();
-      Init(address.first, address.second, client_call_manager_);
+      std::pair<std::string, int> address;
+      int index = 0;
+      for (; index < RayConfig::instance().ping_gcs_rpc_server_max_retries(); ++index) {
+        address = get_server_address_();
+        if (Ping(address.first, address.second, 100)) {
+          RAY_LOG(INFO) << "Target rpc server " << address.first << ":" << address.second
+                        << " is valid.";
```

Comment:
"Reconnected to GCS server: xxx." would be more clear.

Reply:
fixed